### PR TITLE
feat: basedir option added to configuration for generate command

### PR DESCRIPTION
**Pull request by yigitkurtcu**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

Issue Number: [#1849](https://github.com/nestjs/nest-cli/issues/1849)

## What is the new behavior?
Added a new option "baseDir". With this new option, new files related to this directory will be created and imported to the app module.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No

### CODE REVIEW

**Comment by yigitkurtcu** [comment]
Pull request for Schemestore: https://github.com/SchemaStore/schemastore/pull/2967

File: actions/generate.action.ts [hunk 1]
```diff
@@ -125,6 +125,9 @@ const generateFiles = async (inputs: Input[]) => {
     }
   }
 
+  if (configuration.generateOptions?.baseDir)
+    sourceRoot += `\/${configuration.generateOptions?.baseDir}`
```
Comment (kamilmysliwiec) [inline actions/generate.action.ts:129]: we should use `path.join()` here

Comment (yigitkurtcu) [reply actions/generate.action.ts]: Fixed

**Comment by yigitkurtcu** [comment]
> Pull request for Schemestore: [SchemaStore/schemastore#2967](https://github.com/SchemaStore/schemastore/pull/2967)

By the way this pr has already been merged. @kamilmysliwiec

**Review by micalevisk — CHANGES_REQUESTED** [review]
please revert the changes made on `package-lock.json`. I don't think you meant to change that

hint: use `npm ci` instead of `npm install` otherwise may you'll end up touching that file :)

**Comment by micalevisk** [comment]
also @yigitkurtcu woud you like to update the docs site as well? :eyes:

**Comment by Clovel** [comment]
Any news on this ? It would indeed be a useful feature for the NestJS CLI !

**Comment by yigitkurtcu** [comment]
I will update this.

**Pull request closed without merging**